src/store/accounts/index.ts

```diff
@@ -0,0 +1,206 @@
+import { MempoolTransaction, Transaction } from '@blockstack/stacks-blockchain-api-types';
+import { Account, getStxAddress } from '@stacks/wallet-sdk';
+import { atom, selector, waitForAll } from 'recoil';
+import BN from 'bn.js';
+
+import type { AllAccountData } from '@common/api/accounts';
+import { fetchAllAccountData } from '@common/api/accounts';
+import { apiRevalidation, intervalStore } from '@store/common/api-helpers';
+import { fetcher } from '@common/api/wrapped-fetch';
+
+import { transactionRequestStxAddressState } from '@store/transactions/requests';
+import { currentNetworkState } from '@store/networks';
+import { DEFAULT_POLLING_INTERVAL } from '@store/common/constants';
+import { walletState } from '@store/wallet';
+import { transactionNetworkVersionState } from '@store/transactions';
+
+/**
+ * --------------------------------------
+ * Overview
+ * --------------------------------------
+ *
+ * accountsState -- the array of accounts from the wallet state, uses `walletState`
+ * accountsWithAddressState - a mapped array of the above state with the current network mode stx address included
+ * currentAccountIndexStore - the last selected index by the user (or default 0)
+ * hasSwitchedAccountsState - a toggle for when a user switches accounts during a pending transaction request
+ * transactionAccountIndexState - if `stxAccount` is passed with a transaction request, this is the index of that address
+ * currentAccountState - the current active account, either the account associated with a pending transaction request, or the one selected by the user
+ * currentAccountStxAddressState - a selector for the address of the current account
+ * accountDataState -- all external API data for the selected STX address
+ * accountBalancesState - a selector for the balances from `accountDataState`
+ * accountInfoStore - external API data from the `v2/accounts` endpoint, should be the most up-to-date
+ */
+
+enum KEYS {
+  ALL_ACCOUNTS = 'accounts/ALL_ACCOUNTS',
+  ALL_ACCOUNTS_WITH_ADDRESSES = 'accounts/ACCOUNTS_WITH_ADDRESSES',
+  HAS_SWITCHED_ACCOUNTS = 'accounts/HAS_SWITCHED_ACCOUNTS',
+  TRANSACTION_ACCOUNT_INDEX = 'accounts/TRANSACTION_ACCOUNT_INDEX',
+  CURRENT_ACCOUNT_INDEX = 'accounts/CURRENT_ACCOUNT_INDEX',
+  CURRENT_ACCOUNT = 'accounts/CURRENT_ACCOUNT',
+  CURRENT_ACCOUNT_ADDRESS = 'accounts/CURRENT_ACCOUNT_ADDRESS',
+  CURRENT_ACCOUNT_DATA = 'accounts/CURRENT_ACCOUNT_DATA',
+  CURRENT_ACCOUNT_BALANCES = 'accounts/CURRENT_ACCOUNT_BALANCES',
+  CURRENT_ACCOUNT_INFO = 'accounts/CURRENT_ACCOUNT_INFO',
+  CURRENT_ACCOUNT_TRANSACTIONS = 'accounts/CURRENT_ACCOUNT_TRANSACTIONS',
+}
+
+//--------------------------------------
+// All accounts
+//--------------------------------------
```

Comment:
I think we should all agree on a style for documentation in the code so we don't all do it differently?